### ISSUE TITLE
Remove BASECONVRATE from CURRENCYFORMATS table

### ISSUE THREAD

**Issue by slodki** [moneymanagerex/moneymanagerex#1541]
Now we have 2 overlapping implementations of currency rates: one simple inside CURRENCYFORMATS and more sophisticated in dedicated CURRENCYHISTORY table.

My proposal:
* use only CURRENCYHISTORY records
* remove BASECONVRATE field from CURRENCYFORMATS
* do not store meaningless "1" default value in database
* implement following logic to maintain compatibility with current assumptions:
```c
double find_currency_rate_for_given_date(currency curr, date d) {
  if (currencyhistory(curr) is empty || d<oldest in currencyhistory(curr))
    return 1 /* no conversion done if rate is unknown */
  else iterate through currencyhistory(curr) records...
}
```

This should help with #1158, #1183, #1537, improve data model design (removes redundancy) and improve UX.

Problems:
1. Migration from old model for currencies with no CURRENCYHISTORY and with BASECONVRATE <> 1 - there is not enough details available to migrate to new structure. Proposals: a) migrate user's value adding "1900-01-01" date to it b) not migrate BASECONVRATE value to CURRENCYHISTORY
2. Migration from old model BASECONVRATE not matching CURRENCYHISTORY values. Proposals: same as above.

**Comment by gabriele-v** [comment]
Hi @slodki , two implementations are not overlapping:
- baseconvrate if user doesn't wont to take care of currencyhistory (IE download changes every time etc..) and just want a fixed change rate
- currencyhistory if user wants daily change rates

Up to me we could keep both, but better specifying with a tooltip (and maybe renaming to something like "fixedrate") baseconvrate.

**Comment by slodki** [comment]
But when user select one of them we must disable, reset values (?) and ignore second method for currency given. User cannot define both at the same time.

**Comment by slodki** [comment]
We have multiple errors at homepage, reports, stocks, budgets in v1.4.0 related to this. Multi-currency is simply broken now.
We must select right direction for managing currency rates before fixing them.

**Comment by slodki** [comment]
in alpha.4 version only top-left update button writes to BASECONVRATE, bottom-right button updates history only, rate value is displayed in left panel but BASECONVRATE is not updated.

**Comment by gabriele-v** [comment]
Yep, I'll update history to write BASECONVRATE with latest value.

**Comment by gabriele-v** [comment]
I have completed patch for this bug (completely removes BASECONVRATE, I've found a fast way to check nearest change rate), but we need to
- [ ] Apply other PRs before because I need to issue DB v13
- [ ]  @MisterY could you confirm that you can apply this fix to Android version too?

After that, I think no more DB patches are needed for 1.4.0 release, so we need to syncronize Android and Desktop beta release in order to don't leave user with one platform not working. Then syncronize also release of official versions.

**Comment by gabriele-v** [comment]
@MisterY is it ok for you to remove BASECONVRATE?
I've the patch for it ready, considering that we are doing some changes DB side for 1.4.0 I think you could add this too.
Let me know.

**Comment by MisterY** [comment]
Sorry, I was offline for most of the last week. How is it being replaced? Do you have a link to a description or the code? I need to plan when to make these latest mods.
I've recently moved to Linux so still don't have a working dev environment for Android.
